### DISCUSSION TITLE
How to update the values of a Select widget dynamically?

### DISCUSSION THREAD

**Discussion by trixpan** [Textualize/textual#2934] [Q&A] [answered]
I am trying to create a TUI where once the use makes a choice in

```
...
  Select(options=[("choice1", "choice2")], id="widget1")`
...
```

The values of

```
...
  Select(options=[], id="widget2")`
...
```

get populated. 

I created a reactive variable and can use the variable to write to TextLog (for debugging purposes), however trying to use 

```
...
        widget2_selector = self.query_one("#widget2")
        widget2_selector.options = self.reactive_options
        widget2_selector.refresh()
...
```

Doesn't seem to work.

Instead I need to find the parent, remove the Select widget, create a new one, mount and refresh:

```
...
        old_selector = self.query_one("#widget2")
        parent = old_selector.parent
        old_selector.remove()
        new_selector = Select(options=self.reactive_options, id="widget2")
        parent.mount(new_selector)
        parent.refresh()
...
```

**Comment by davep** [comment, marked as answer]
Looks like you're looking for set_options: https://textual.textualize.io/widgets/select/#textual.widgets._select.Select.set_options

**Reply by trixpan** [reply]
Epic!  

BTW, the framework is incredible. I never wrote a TUI and I am mind blown with how simple it is to get the "basics" working.

**Reply by davep** [reply]
Glad to hear you're enjoying it. 🙂
